internal/provider_framework/provider.go

```diff
@@ -0,0 +1,213 @@
+package provider_framework
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+	"github.com/hashicorp/terraform-plugin-framework/tfsdk"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+type provider struct {
+	proxyURL     *url.URL
+	proxyFromEnv bool
+}
+
+// Enforce interfaces we want provider to implement
+var _ tfsdk.Provider = (*provider)(nil)
+
+func New() tfsdk.Provider {
+	return &provider{
+		proxyURL:     nil,
+		proxyFromEnv: true,
```

Comment:
Can you return the struct with zero values:
```go
return &provider{}
```

Reply:
Unfortunately no, I need the `proxyFromEnv: true`.
This is because there is no way to impose a default value on Provider configuration, so what I do is to store it in the structure that holds the value that will be consumed later on, and if the logic doesn't set it, then `true` is the desired default value.